src/installer/tests/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/NetCoreApp3CompatModeTests.cs

```diff
@@ -0,0 +1,91 @@
+        {
+            sharedTestState = fixture;
+        }
+
+        [Fact]
+        public void Bundle_Is_Extracted()
+        {
+            var fixture = sharedTestState.TestFixture.Copy();
+            string singleFile;
+            Bundler bundler = BundleHelper.BundleApp(fixture, out singleFile, BundleOptions.BundleAllContent);
+            var extractionBaseDir = BundleHelper.GetExtractionRootDir(fixture);
+
+            Command.Create(singleFile, "executing_assembly_location trusted_platform_assemblies assembly_location System.Console")
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .EnvironmentVariable(BundleHelper.DotnetBundleExtractBaseEnvVariable, extractionBaseDir.FullName)
+                .Execute()
+                .Should()
+                .Pass()
+                // Validate that the main assembly is running from disk (and not from bundle)
+                .And.HaveStdOutContaining("ExecutingAssembly.Location: " + extractionBaseDir.FullName)
+                // Validate that TPA contains at least one framework assembly from the extraction directory
+                .And.HaveStdOutContaining("System.Runtime.dll")
+                // Validate that framework assembly is actually loaded from the extraction directory
+                .And.HaveStdOutContaining("System.Console location: " + extractionBaseDir.FullName);
+
+            var extractionDir = BundleHelper.GetExtractionDir(fixture, bundler);
+            var bundleFiles = BundleHelper.GetBundleDir(fixture).GetFiles().Select(file => file.Name).ToArray();
+            var publishedFiles = Directory.GetFiles(BundleHelper.GetPublishPath(fixture), searchPattern: "*", searchOption: SearchOption.AllDirectories)
+                .Select(file => Path.GetFileName(file))
+                .Except(bundleFiles)
+                .ToArray();
+            var bundlerFiles = BundleHelper.GetBundleDir(fixture).GetFiles();
+            extractionDir.Should().HaveFiles(publishedFiles);
+        }
+
+        public class SharedTestState : IDisposable
+        {
+            public TestProjectFixture TestFixture { get; set; }
+            public RepoDirectoriesProvider RepoDirectories { get; set; }
+
+            public SharedTestState()
+            {
+                RepoDirectories = new RepoDirectoriesProvider();
+                TestFixture = new TestProjectFixture("SingleFileApiTests", RepoDirectories);
+                TestFixture
+                    .EnsureRestoredForRid(TestFixture.CurrentRid, RepoDirectories.CorehostPackages)
+                    .PublishProject(runtime: TestFixture.CurrentRid, outputDirectory: BundleHelper.GetPublishPath(TestFixture));
+
+                // The above will publish the app as self-contained using the stock .NET Core SDK (typically from Program Files)
+                // In order to test the hosting components we need the hostpolicy and hostfxr built by the repo
```

Comment:
Oh, wait, these tests are not using `singlefilehost`? Is that intentional? It seems like - especially since this is specifically for backcompat - we should be using the new single-file host.

Reply:
I'm looking into this - it's broken in more than one way unfortunately. The way the tests work was viable in 3.1, but it's not going to work in 5.0 at all.